### PR TITLE
completed week 2 assessments

### PR DESCRIPTION
**Pull request by cesarcuadro**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅  ] Coding Challenges
- [ ✅ ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

If possible, can you comment if I created an edge case correctly on the first coding challenge question. Also on the third coding challenge, if I was able to properly explain my method of using regex.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,13 +10,26 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
+/*
+Boiler Plate for Test Function:
```
Comment (elysemontano) [inline code-challenges.test.js:14]: Nicely done!

File: code-challenges.test.js [hunk 2]
```diff
@@ -25,35 +38,140 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const object1 = { number: 15 }
+describe("divByThree", () => {
+    it("takes a object as an argument and decides if the number inside it is evenly divisible by three or not.", () => {
+    const object1 = { number: 15 }
+    const object1OutPut = "15 is divisible by three"
+    expect(divByThree(object1.number)).toEqual(object1OutPut)
```
Comment (elysemontano) [inline code-challenges.test.js:45]: An important thing to note in your tests, you don't want to perform any logic or data manipulation inside here.  That should be reserved for the function.

File: code-challenges.test.js [hunk 3]
```diff
@@ -25,35 +38,140 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+        return `${object} is divisible by three`
+    } else if (object % 3 !== 0) {
+        return `${object} is not divisible by three`
+    } else {
+        return `${object} is not a number`
+    }
+}
+
+// Originally I was getting a syntax error and that the input of the object was undefined.
+//In order to get the correct input I needed, I used dot notation in order to access the value from the key: number.
```
Comment (elysemontano) [inline code-challenges.test.js:85]: As mentioned above, that process should be handled inside of your function and not inside of your test.  Otherwise, this looks excellent!

File: code-challenges.test.js [hunk 4]
```diff
@@ -25,35 +38,140 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+// ReferenceError: capArray is not defined
+
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
+/*
+Create a function that takes uses one parameter to input an array
```
Comment (elysemontano) [inline code-challenges.test.js:117]: Beautiful pseudocode!

File: code-challenges.test.js [hunk 5]
```diff
@@ -25,35 +38,140 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
 // Expected output: 2
 
 // b) Create the function that makes the test pass.
+
+/*
+create a function that will take in one parameter as an argument
+A string will be passed through the parameter of the function
+Iterate through a string by using regex and look for characters that are vowels 
+    i.e."a", "e", "i", "o", "u"
+    Regex , otherwise known as regular expressions, are patterns to match character combinations in a string. In this case, we are searching for the characters that match vowels. while the 'i' on the outside  of the matchers is a method in which makes the characters case insensitive. Meaning that we can match the characters in either lowercase or uppercase.
```
Comment (elysemontano) [inline code-challenges.test.js:167]: Excellent definition of Regex!

File: coding-concepts.js [hunk 6]
```diff
@@ -65,4 +68,12 @@ const learnStudent = new Learn("George")
...
-// b) Verify and explain:
+/*
+Learn
+{
+  student: "George",
+  cohort: "Foxtrot",
+  year: 2022
+}
+*/
+// b) Verify and explain: inside the variable, the keyword 'new' creates a new object through the class 'Learn' and passes through the parameter of name as "George". So logging the variable 'learnStudent', the console will print the new object 'Learn' with the key:value pairs of -> student: "George", cohort: "Foxtrot", and year: 2022.
```
Comment (elysemontano) [inline coding-concepts.js:79]: Wonderful work in this section!

File: interview-questions.md [hunk 7]
```diff
@@ -8,50 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Your answer:
+Your answer: The difference between a parameter and an argument is that a parameter has a value that passes through an argument. 
+For example when writing a function through JavaScript
 
-Researched answer:
+const functionName = (parameter) => {
+    return parameter
+}
+
+console.log(functionName(argument))
```
Comment (elysemontano) [inline interview-questions.md:18]: Examples are such a great way to communicate concepts.  Great work!

File: interview-questions.md [hunk 8]
```diff
@@ -8,50 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Your answer:
+Your answer: The main difference between map and filter is that map keeps the original length of the array while filter only takes in the elements that fulfill its arguments(i.e. creates a new array with a smaller length than the original array)
 
-Researched answer:
+Researched answer: Map accesses the array and changes the values of each element, but it keeps the original length of the array. Filter uses boolean conditions in order to return a new array, if the element returns to true then it keeps the element, if it returns false then it excludes the element.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is when you go through each element of an array, or a character in a string, until a defined condition is met.
```
Comment (elysemontano) [inline interview-questions.md:44]: A good way to expand this answer would be to talk about some of the different types of iteration you have used.

**Review by elysemontano — APPROVED** [review]
Excellent work Cesar!  I left some feedback throughout for you to take a look at.
The current assessment level is competent - Some things to pay attention to that may drop your assessment level down in the future would be keeping all logic within your function, not in your tests.  Beyond that, almost all of your interview answers were excellent, however, there were a few that it seemed like you held back.  Make sure to keep that momentum up on all the problems.
